### PR TITLE
refactoring for import csv

### PR DESCRIPTION
**Pull request by EvanBldy**
**Problem**
- for assets/shots/edits imports for example description is deleted when we do an update and the column is missing 
- the code sometimes is hard to understand (for tasks update of assets/shots/edits for example)
**Solution**
- assets/shots/edits/persons refactoring of the code + check all behaviours to stop deleting informations

### CODE REVIEW

File: zou/app/blueprints/source/csv/assets.py [hunk 1]
```diff
@@ -52,14 +56,83 @@ def prepare_import(self, project_id):
...
+        for task_type in self.task_types_in_project_for_assets:
+            task_status_name = row.get(task_type.name, None)
+            task_status_id = None
+            if task_status_name is not None:
+                for status_id, status_names in self.task_statuses.items():
+                    if task_status_name in status_names:
+                        task_status_id = status_id
+                        break
+                if task_status_id is None:
+                    raise TaskStatusNotFoundException(
```
Comment (frankrousseau) [inline zou/app/blueprints/source/csv/assets.py:70]: Be careful, raising exceptions like this generates a 404 error.

File: zou/app/blueprints/source/csv/assets.py [hunk 2]
```diff
@@ -68,6 +141,10 @@ def import_row(self, row, project_id):
                     "id"
                 ]
             episode_id = self.episodes.get(episode_name, None)
+        elif episode_name is not None:
+            raise ImportRowException(
```
Comment (frankrousseau) [inline zou/app/blueprints/source/csv/assets.py:145]: Same here. You can't raise exception in the midlle of the code without handling them somewhere.

**Comment by frankrousseau** [comment]
Tests are broken. Could you fix them and add some more?

File: zou/app/blueprints/source/csv/assets.py (hunk 1 above)
Comment (EvanBldy) [reply zou/app/blueprints/source/csv/assets.py:70]: Effectively, I have to use ImportRowException

File: zou/app/blueprints/source/csv/assets.py (hunk 2 above)
Comment (EvanBldy) [reply zou/app/blueprints/source/csv/assets.py]: I use RowException now that is handled in the main class

**Pull request merged**